### DISCUSSION TITLE
reranking model use API

### DISCUSSION THREAD

**Discussion by edisonzf2020** [open-webui/open-webui#3317] [Ideas]
**Is your feature request related to a problem? Please describe.**
I hope the reranking model uses an API, because there are already several excellent rerank model APIs available.
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by oatmealm** [comment]
Any news on this? I'm running openwebui on an rpi5 and would love to not have the re-ranker kill my server. Jina's re-ranker could be useful as an option while ollama is missing this feature.

**Comment by bgeneto** [comment]
Yep. It would be nice to use, for example, voyageai.com or jina.ai reranking API models additionally to its embedding models.

**Comment by cesekjones** [comment]
+1 on this. 

LiteLLM can be setup as a Cohere API compatible rerank proxy. If the Cohere API was added to Open WebUI as an option for reranking, then LiteLLM could be used to give access to pretty much any reranker. In much the same way that the OpenAI API compatibility means Open WebUI + LiteLLM gives access to pretty much any endpoint LLM.

**Comment by SlavikCA** [comment]
reranking is not yet supported by Ollama.

I see that there is PR: https://github.com/ollama/ollama/pull/7219

But it's kind stuck...

**Comment by Marcel2508** [comment]
Whats the current state of this? llama.cpp nowadays supports a reranking-endpoint: https://github.com/ggerganov/llama.cpp/pull/9510

**Comment by tan-yong-sheng** [comment]
+1 on this

> LiteLLM can be setup as a Cohere API compatible rerank proxy. If the Cohere API was added to Open WebUI as an option for reranking, then LiteLLM could be used to give access to pretty much any reranker. In much the same way that the OpenAI API compatibility means Open WebUI + LiteLLM gives access to pretty much any endpoint LLM.

I am thinking to do the same setup as well.

**Comment by unclemusclez** [comment]
i would also prefer to run this off of the gpu with llama.cpp

**Comment by ArtrixTech** [comment]
would love this too.
